nixos/modules/services/display-managers/sddm.nix

```diff
@@ -234,9 +247,9 @@ in
 
     assertions = [
       {
-        assertion = xcfg.enable;
+        assertion = xcfg.enable || cfg.wayland.enable;
         message = ''
-          SDDM requires services.xserver.enable to be true
+          SDDM requires either services.xserver.enable or services.displayManager.sddm.wayland.enable to be true
         '';
       }
```

Comment:
This now introduces a footgun where the user can set `services.displayManager.sddm.enable` but not `services.displayManager.enable` and wonder why nothing works. Either the former should enable the latter or the assertion should be modified.

Reply:
I don't understand. There is no option named `services.displayManager.enable` or does this line mention it.

Reply:
You defined this option in `nixos/modules/services/display-managers/default.nix`. Without setting that to true as well the generated display-manager systemd service has a broken unit file and can't even start. I marked this line because one option is adjusting the assertion.

Reply:
You're right. Either my fork is already getting out of sync or I did a bad rebase but on my fork this was set to true.

But in general there is still another PR required to be able to truly turn of xserver when using plasma 6 without enabling extra bits and keeping thins running.